src/main.py

```diff
@@ -0,0 +1,80 @@
+class Product:
+    """Класс продукта для интернет-магазина"""
+
+    name: str
+    description: str
+    price: float
+    quantity: int
+
+    def __init__(self, name: str, description: str, price: float, quantity: int):
+        """Конструктор класса Product для создания экземпляра объекта"""
+        self.name = name
+        self.description = description
+        self.price = price
+        self.quantity = quantity
+
+
+class Category:
+    """Класс категории товара для интернет-магазина"""
+
+    name: str
+    description: str
+    products: list
+    category_count: int = 0
+    product_count: int = 0
+
+    def __init__(self, name: str, description: str, products: list | None = None):
```

Comment:
Тип продуктов лучше указать явно
```suggestion
    def __init__(self, name: str, description: str, products: list[Product] | None = None):
```